tests/Integration/IntegrationTest.php

```diff
@@ -0,0 +1,206 @@
+use function trim;
+
+class IntegrationTest extends ApplicationTestCase
+{
+    protected function setUp(): void
+    {
+        setlocale(LC_ALL, 'en_US.utf8');
+    }
+
+    /** @param string[] $compareFiles */
+    #[DataProvider('getTestsForDirectoryTest')]
+    public function testHtmlIntegration(
+        string $inputPath,
+        string $expectedPath,
+        string $outputPath,
+        array $compareFiles,
+    ): void {
+        system('rm -rf ' . escapeshellarg($outputPath));
+        self::assertDirectoryExists($inputPath);
+        self::assertDirectoryExists($expectedPath);
+        self::assertNotEmpty($compareFiles);
+
+        $skip = file_exists($inputPath . '/skip');
+        $configurationFile = null;
+        if (file_exists($inputPath . '/guides.xml')) {
+            $configurationFile = $inputPath . '/guides.xml';
+        }
+
+        try {
+            system('mkdir ' . escapeshellarg($outputPath));
+
+            $this->prepareContainer($configurationFile);
+            $command = $this->getContainer()->get(Run::class);
+            assert($command instanceof Run);
+
+            $input = new ArrayInput(
+                [
+                    'input' => $inputPath,
+                    'output' => $outputPath,
+                    '--theme' => 'typo3docs',
+                    '--log-path' => $outputPath . '/logs',
+                    '--output-format' => ['html', 'intersphinx'],
+                ],
+                $command->getDefinition(),
+            );
+
+            $outputBuffer = new BufferedOutput();
+
+            $command->run(
+                $input,
+                $outputBuffer,
+            );
+            if (!file_exists($expectedPath . '/logs/error.log')) {
+                self::assertFileDoesNotExist($outputPath . '/logs/error.log');
+            }
+
+            if (!file_exists($expectedPath . '/logs/warning.log')) {
+                self::assertFileDoesNotExist($outputPath . '/logs/warning.log');
+            }
+
+            foreach ($compareFiles as $compareFile) {
+                $outputFile = str_replace($expectedPath, $outputPath, $compareFile);
+                if (str_ends_with($compareFile, '.log')) {
+                    self::assertFileContainsLines($compareFile, $outputFile);
+                } else {
+                    self::assertFileEqualsTrimmed($compareFile, $outputFile, 'Expected file path: ' . $compareFile);
+                }
+            }
+        } catch (ExpectationFailedException $e) {
+            if ($skip) {
+                self::markTestIncomplete(file_get_contents($inputPath . '/skip') ?: '');
+            }
+
+            throw $e;
+        }
+
+        self::assertFalse($skip, 'Test passes while marked as SKIP.');
+    }
+
+    /**
+     * Asserts that each line of the expected file is contained in actual
+     *
+     * @throws ExpectationFailedException
+     */
+    public static function assertFileContainsLines(string $expected, string $actual): void
```

Comment:
Why public?

Reply:
doesn't need to be public as it is no data provider